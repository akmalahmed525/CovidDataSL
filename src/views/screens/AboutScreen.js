// @flow
import React, {Component, useCallback} from 'react';
import {View, Text, Alert, Linking, StyleSheet, Dimensions} from 'react-native';
import {connect} from 'react-redux';
import {human} from 'react-native-typography';
import Config from 'react-native-config';

import {t} from 'i18n';
import themes from 'themes';
import {locale} from 'i18n-js';

const {width} = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: themes.colors.white
  },
  banner: {
    flexGrow: 1,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    padding: themes.dimens.bannerPadding
  },
  animationStyle: {
    width: width * 0.3
  },
  textStyleTitle1: {
    textAlign: 'center',
    fontFamily: 'Rubik-Regular'
  },
  textStyleTitle2: {
    textAlign: 'center',
    fontFamily: 'Rubik-Light'
  },
  linkStyle: {
    color: themes.colors.blue
  }
});

type URLLinkProps = {
  url: string,
  style: Object | Array<any>,
  children: any
};
const OpenURLButton = ({url, style, children}: URLLinkProps) => {
  const handlePress = useCallback(async () => {
    const supported = await Linking.canOpenURL(url);

    if (supported) {
      await Linking.openURL(url);
    } else {
      Alert.alert(`Don't know how to open this URL: ${url}`);
    }
  }, [url]);

  return (
    <Text style={style} onPress={handlePress}>
      {children}
    </Text>
  );
};

type Props = {};
class AboutScreenView extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.banner}>
          <Text style={[human.title1, styles.textStyleTitle1]}>
            {t('about.title')}
          </Text>
          <Text style={[human.body, styles.textStyleTitle2]} />
          <Text style={[human.title2, styles.textStyleTitle2]}>
            {t('about.description')}
          </Text>
          <Text style={[human.body, styles.textStyleTitle2]} />
          <OpenURLButton
            style={[human.body, styles.textStyleTitle1, styles.linkStyle]}
            url={Config.REPOSITORY_URL}>
            {t('about.get_code', locale)}
          </OpenURLButton>
        </View>
      </View>
    );
  }
}

const mapStateToProps = (state) => ({state});

const AboutScreen = connect<Props, *, *, *, *, *>(
  mapStateToProps,
  {}
)(AboutScreenView);
export default AboutScreen;
