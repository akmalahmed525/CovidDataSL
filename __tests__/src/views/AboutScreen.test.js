import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import AboutScreen from 'src/views/screens/AboutScreen';

const mockStore = configureStore([]);

describe('About screen', () => {
  let store;
  let aboutScreen;

  beforeEach(() => {
    store = mockStore({});
    aboutScreen = renderer.create(
      <Provider store={store}>
        <AboutScreen />
      </Provider>
    );
  });

  test('Must match the snapshot', () => {
    expect(aboutScreen.toJSON()).toMatchSnapshot();
  });
});
