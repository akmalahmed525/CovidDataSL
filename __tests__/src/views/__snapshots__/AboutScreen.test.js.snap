// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About screen Must match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "flexGrow": 1,
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 28,
            "fontWeight": "400",
            "letterSpacing": 0.355469,
            "lineHeight": 34,
          },
          Object {
            "fontFamily": "Rubik-Regular",
            "textAlign": "center",
          },
        ]
      }
    >
      Covid-19 Data Visualization Application
    </Text>
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "400",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          },
          Object {
            "fontFamily": "Rubik-Light",
            "textAlign": "center",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0.34375,
            "lineHeight": 28,
          },
          Object {
            "fontFamily": "Rubik-Light",
            "textAlign": "center",
          },
        ]
      }
    >
      An open source React Native application developed to interact with Covid-19 data API of Sri Lankan Health Promotion Bureau
    </Text>
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "400",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          },
          Object {
            "fontFamily": "Rubik-Light",
            "textAlign": "center",
          },
        ]
      }
    />
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "400",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          },
          Object {
            "fontFamily": "Rubik-Regular",
            "textAlign": "center",
          },
          Object {
            "color": "#0000EE",
          },
        ]
      }
    >
      Get the source code
    </Text>
  </View>
</View>
`;
